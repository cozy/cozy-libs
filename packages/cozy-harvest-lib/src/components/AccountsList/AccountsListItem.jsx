import React from 'react'
import PropTypes from 'prop-types'

import { Account } from 'cozy-doctypes'
import Card from 'cozy-ui/transpiled/react/Card'
import MidEllipsis from 'cozy-ui/transpiled/react/MidEllipsis'
import { Caption } from 'cozy-ui/transpiled/react/Text'

import Status from './Status'

export class AccountsListItem extends React.PureComponent {
  render() {
    const { account, onClick, konnector, trigger } = this.props
    const accountName = Account.getAccountName(account)
    const accountLogin = Account.getAccountLogin(account)
    /**
     * When the account is from an OAuth, we cat not have an accountLogin.
     * So have to check if accountLogin is not undefined before doing the check
     *  */
    const shouldShowAccountLogin = accountLogin && accountName !== accountLogin
    return (
      <Card
        className="u-flex u-flex-justify-between u-flex-items-center u-c-pointer u-maw-100"
        onClick={onClick}
      >
        <div className="u-flex-grow-1 u-flex-shrink-1 u-ov-hidden u-mr-1">
          <MidEllipsis text={accountName} />
          {shouldShowAccountLogin && (
            <Caption>
              <MidEllipsis text={accountLogin} />
            </Caption>
          )}
        </div>
        <Status konnector={konnector} trigger={trigger} />
      </Card>
    )
  }
}

AccountsListItem.propTypes = {
  account: PropTypes.object.isRequired,
  konnector: PropTypes.shape({
    name: PropTypes.string,
    vendor_link: PropTypes.string
  }).isRequired,
  trigger: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
}

export default AccountsListItem
