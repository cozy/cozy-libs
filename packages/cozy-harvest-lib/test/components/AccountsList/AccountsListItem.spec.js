import React from 'react'
import { shallow } from 'enzyme'

import AccountsListItem from 'components/AccountsList/AccountsListItem'

describe('AccountsListItem', () => {
  it('should not render the caption since accountLogin is undefined', () => {
    const wrapper = shallow(
      <AccountsListItem
        account={{
          _id: 'account-1'
        }}
        konnector={{
          name: 'test-konnector',
          vendor_link: 'test konnector link'
        }}
        onClick={jest.fn()}
        trigger={jest.fn()}
      />
    )
    const component = wrapper.shallow()
    expect(component.getElement()).toMatchSnapshot()
  })

  it('should render the caption since accountName !== login', () => {
    const wrapper = shallow(
      <AccountsListItem
        account={{
          _id: 'account-1',
          auth: { login: 'mylogin', accountName: 'myAccountName' }
        }}
        konnector={{
          name: 'test-konnector',
          vendor_link: 'test konnector link'
        }}
        onClick={jest.fn()}
        trigger={jest.fn()}
      />
    )
    const component = wrapper.shallow()
    expect(component.getElement()).toMatchSnapshot()
  })
})
