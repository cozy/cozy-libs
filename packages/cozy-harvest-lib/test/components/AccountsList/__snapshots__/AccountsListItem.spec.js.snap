// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsListItem should not render the caption since accountLogin is undefined 1`] = `
<div
  className="styles__c-card___YgP7B u-flex u-flex-justify-between u-flex-items-center u-c-pointer u-maw-100"
  onClick={[MockFunction]}
>
  <div
    className="u-flex-grow-1 u-flex-shrink-1 u-ov-hidden u-mr-1"
  >
    <MidEllipsis
      text="account-1"
    />
  </div>
  <Wrapper
    konnector={
      Object {
        "name": "test-konnector",
        "vendor_link": "test konnector link",
      }
    }
    trigger={[MockFunction]}
  />
</div>
`;

exports[`AccountsListItem should render the caption since accountName !== login 1`] = `
<div
  className="styles__c-card___YgP7B u-flex u-flex-justify-between u-flex-items-center u-c-pointer u-maw-100"
  onClick={[MockFunction]}
>
  <div
    className="u-flex-grow-1 u-flex-shrink-1 u-ov-hidden u-mr-1"
  >
    <MidEllipsis
      text="myAccountName"
    />
    <Caption
      ellipsis={false}
      tag="div"
    >
      <MidEllipsis
        text="mylogin"
      />
    </Caption>
  </div>
  <Wrapper
    konnector={
      Object {
        "name": "test-konnector",
        "vendor_link": "test konnector link",
      }
    }
    trigger={[MockFunction]}
  />
</div>
`;
